bundles/org.openhab.binding.boschspexor/src/main/java/org/openhab/binding/boschspexor/internal/api/service/BoschSpexorBridgeConfig.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.boschspexor.internal.api.service;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * Main settings.
+ *
+ * @author Marc Fischer - Initial contribution.
+ */
+@NonNullByDefault
+public class BoschSpexorBridgeConfig {
+
+    /**
+     * The host name of the Bosch spexor backend.
+     */
+    private String host = "https://api.spexor-bosch.com/";
+
+    /**
+     * The OAuth2 client id
+     */
+    private String clientId = "spexor-3rdparty-service-auth";
+    /**
+     * The OAuth2 scope
+     */
+    private String scope = "3rdparty-service";
+    /**
+     * Interval (in seconds) at which state updates are polled.
+     */
+    private int refreshInterval = 60; // Default: Every 1min
+
+    /**
+     * The token path to request the OAuth2 token.
+     */
+    private String tokenUrl = "api/public/token";
+    /**
+     * The refresh token path to request the OAuth2 token.
+     */
+    private String refreshUrl = "api/public/refresh";
+    /**
+     * The token path to request the OAuth2 token.
+     */
+    private String authorizationUrl = "api/public/auth";
+
+    public String getHost() {
+        return host;
+    }
+
+    public String getClientId() {
+        return clientId;
+    }
+
+    public String getScope() {
+        return scope;
+    }
+
+    public int getRefreshInterval() {
+        return refreshInterval;
+    }
+
+    public String getRefreshUrl() {
+        return refreshUrl;
+    }
+
+    public String getTokenUrl() {
+        return tokenUrl;
+    }
+
+    public String getAuthorizationUrl() {
+        return authorizationUrl;
+    }
+
+    public String buildTokenUrl() {
+        return String.join("", getHost(), tokenUrl);
```

Comment:
well do to my old development experience it was mostly a mess in performance, but I stopped with Java 6 :-) I was more used to avoid + concatenation and mainly used to call such functions. I've have no big deal to change it. [string join vs string concat](https://stackoverflow.com/questions/31817290/string-join-vs-other-string-concatenation-operations)
I will change it.